adapters/deeplabv3/model_adapter.py

```diff
@@ -0,0 +1,480 @@
+
+            Virtual method - need to implement
+
+        :param batch: `np.ndarray`
+        :return: `list[dl.AnnotationCollection]` each collection is per each image / item in the batch
+        """
+        self.model.eval()
+
+        # input_size = self.configuration.get('input_size', 224)
+
+        preprocess = torchvision.transforms.Compose([torchvision.transforms.ToPILImage(),
+                                                     # torchvision.transforms.Resize((input_size, input_size)),
+                                                     self.gray_to_rgb,
+                                                     torchvision.transforms.ToTensor(),
+                                                     torchvision.transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                                                                                      std=[0.229, 0.224, 0.225])])
+
+        batch_annotations = list()
+
+        for img in batch:
+            img_tensor = preprocess(img.astype('uint8')).unsqueeze(0).to(self.device)  # Add batch dimension
+            collection = self.inference(img_tensor)
+            batch_annotations.append(collection)
+
+        return batch_annotations
+
+    def inference(self, img_tensor):
+        labels = list(self.model_entity.id_to_label_map.values())
+        threshold = self.configuration.get('conf_threshold', 0.8)
+
+        with torch.no_grad():  # Forward pass through the model
+            img_output = self.model(img_tensor)['out'][0]
+        probs = torch.softmax(img_output, dim=0)
+        output_predictions = probs.argmax(dim=0)
+
+        # Get the unique class indices in the predictions excluding class index 0
+        unique_class_indices = torch.unique(output_predictions.flatten())
+        unique_class_indices = unique_class_indices[unique_class_indices != 0]
+
+        collection = dl.AnnotationCollection()
+        for class_idx in unique_class_indices:
+            confidence = probs[class_idx].cpu().numpy().max()
+            if confidence < threshold:  # Skip if confidence is below the threshold
+                logger.info(f"Confidence:{confidence} , is lower than threshold: {threshold}")
+                continue
+
+            class_label = labels[class_idx]
+            class_mask = (output_predictions == class_idx).cpu().numpy().astype(np.uint8)
+            contours = self.extract_contours(class_mask)
+            # self.plot_contours(img_tensor.squeeze(0).permute(1, 2, 0).cpu().numpy(), contours[0])
+
+            for polygon in contours:
+                collection.add(annotation_definition=dl.Polygon(geo=polygon, label=class_label),
+                               model_info={'name': self.model_entity.name,
+                                           'confidence': confidence,
+                                           'model_id': self.model_entity.id})
+        return collection
+
+    def convert_from_dtlpy(self, data_path, **kwargs):
```

Comment:
If it's going to be empty, I don't think we must leave it here

Reply:
it raise if the function is not implemented